datastore/gcloud/aio/datastore/value.py

```diff
@@ -24,33 +28,60 @@ def __repr__(self) -> str:
 
     @classmethod
     def from_repr(cls, data: Dict[str, Any]) -> 'Value':
-        for value_type, type_name in TYPES.items():
+        supported_types = cls._get_supported_types()
+
+        for value_type, type_name in supported_types.items():
             json_key = type_name.value
             if json_key in data:
                 if json_key == 'nullValue':
                     value = None
-                elif value_type == datetime:  # no parsing needed
-                    value = data[json_key]
+                elif value_type == datetime:
+                    value = datetime.strptime(data[json_key],
+                                              '%Y-%m-%dT%H:%M:%S.%f000Z')
+                elif value_type == cls.key_kind:
+                    value = cls.key_kind.from_repr(data[json_key])
                 else:
                     value = value_type(data[json_key])
                 break
         else:
-            supported_types = [type_name.value for type_name in TYPES.values()]
+            supported = [name.value for name in supported_types.values()]
             raise NotImplementedError(
                 f'{data.keys()} does not contain a supported value type'
-                f' (any of: {supported_types})')
+                f' (any of: {supported})')
 
-        exclude_from_indexes = bool(data['excludeFromIndexes'])
+        # Google may not populate that field. This can happen with both
+        # indexed and non-indexed fields.
+        exclude_from_indexes = bool(data.get('excludeFromIndexes', False))
```

Comment:
There's more than one solution here. I don't think it matters much when parsing, so defaulting to False seems sensible.